core/trino-main/src/test/java/io/trino/operator/aggregation/state/TestLongDecimalWithOverflowAndLongStateSerializer.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.aggregation.state;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.VariableWidthBlockBuilder;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestLongDecimalWithOverflowAndLongStateSerializer
+{
+    private static final LongDecimalWithOverflowAndLongStateFactory STATE_FACTORY = new LongDecimalWithOverflowAndLongStateFactory();
+
+    @Test(dataProvider = "input")
+    public void testSerde(long low, long high, long overflow, long count)
+    {
+        LongDecimalWithOverflowAndLongState state = STATE_FACTORY.createSingleState();
+        state.getDecimalArray()[0] = high;
+        state.getDecimalArray()[1] = low;
+        state.setOverflow(overflow);
+        state.setLong(count);
+        state.setNotNull();
+
+        LongDecimalWithOverflowAndLongState outState = roundTrip(state);
+
+        assertTrue(outState.isNotNull());
+        assertEquals(outState.getDecimalArray()[0], high);
+        assertEquals(outState.getDecimalArray()[1], low);
+        assertEquals(outState.getOverflow(), overflow);
+        assertEquals(outState.getLong(), count);
+    }
+
+    @Test
+    public void testNullSerde()
+    {
+        // state is created null
+        LongDecimalWithOverflowAndLongState state = STATE_FACTORY.createSingleState();
+
+        LongDecimalWithOverflowAndLongState outState = roundTrip(state);
+
+        assertFalse(outState.isNotNull());
+    }
+
+    private LongDecimalWithOverflowAndLongState roundTrip(LongDecimalWithOverflowAndLongState state)
+    {
+        LongDecimalWithOverflowAndLongStateSerializer serializer = new LongDecimalWithOverflowAndLongStateSerializer();
+        BlockBuilder out = new VariableWidthBlockBuilder(null, 1, 0);
+
+        serializer.serialize(state, out);
+
+        Block serialized = out.build();
+        LongDecimalWithOverflowAndLongState outState = STATE_FACTORY.createSingleState();
+        serializer.deserialize(serialized, 0, outState);
+        return outState;
+    }
+
+    @DataProvider
+    public Object[][] input()
+    {
+        return new Object[][] {
+                {1, 0, 0, 1},
```

Comment:
you should use different values than `1` for each column to make sure two values are not mismatched